### PR TITLE
[Feature][DataObject] add title to class definition

### LINKED ISSUE #12647: Possibility to add an title to classdefinitions
### Feature description

Currently there is only a name for classdefinitions. With the name the classfile is generated and the name is currently used for translating the class in the backend.
It would be nice to have an additional field (title) where you can add a translation key like app.classes.product which is used for translations in your translation file.
If no title is available, the behavior stays the same.

I would provide a PR for this :)

### PR DESCRIPTION
**Pull request by dpfahlbusch**
## Changes in this pull request  
Resolves #12647

### CODE REVIEW

File: models/DataObject/ClassDefinition.php [hunk 1]
```diff
@@ -51,6 +51,13 @@ final class ClassDefinition extends Model\AbstractModel
      */
     public $name;
 
+    /**
+     * @internal
+     *
+     * @var string
+     */
+    public $title = '';
```
Comment (blankse) [inline models/DataObject/ClassDefinition.php]:
```suggestion
    public string $title = '';
```

File: models/DataObject/ClassDefinition.php [hunk 2]
```diff
@@ -1255,6 +1266,26 @@ public function getDescription()
         return $this->description;
     }
 
+    /**
+     * @param string $title
+     *
+     * @return $this
+     */
+    public function setTitle($title)
```
Comment (blankse) [inline models/DataObject/ClassDefinition.php]:
```suggestion
    public function setTitle(string $title): static
```

File: models/DataObject/ClassDefinition.php [hunk 3]
```diff
@@ -1255,6 +1266,26 @@ public function getDescription()
...
+    {
+        $this->title = $title;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getTitle()
```
Comment (blankse) [inline models/DataObject/ClassDefinition.php]:
```suggestion
    public function getTitle(): string
```

**Review by blankse — COMMENTED** [review]
We should add typehints for new attributes/methods

**Comment by dpfahlbusch** [comment]
Wasn't sure if this is already a thing :)
Thanks @blankse

**Comment by dpfahlbusch** [comment]
Ready for review @fashxp

**Comment by robertSt7** [comment]
@dpfahlbusch  Thanks for the PR.  It looks good, but could you also please add some documentation like: Add "title" field to classdefinitions, where you can add a translation key like app.classes.product which is used for translations in your translation file. Thank you

And what do you think about the use of the title in the following menu?
Screenshot (4)

**Comment by dpfahlbusch** [comment]
@robertSt7 I will add that :)

**Comment by dpfahlbusch** [comment]
@robertSt7 added documentation and translated the dropdown, so please review again :)

**Comment by robertSt7** [comment]
@dpfahlbusch Thanks a lot for the PR :)

**Pull request merged**